client/templates/courses/tagged-courses.js

```diff
@@ -1,8 +1,49 @@
+Template.taggedCourses.onCreated(function(){
+  // Get reference to template instance
+  var instance = this;
+
+  // Accessing the Iron.controller to invoke getParams method of Iron Router.
+  var router = Router.current();
+
+  // Getting Params of the URL
+  instance.tag = router.params.tag;
+
+  // Subscribe to courses tagged with the current tag
+  instance.subscribe('taggedCourses', instance.tag);
+
+  // Subscribe to course images
+  instance.subscribe('images');
+});
+
+
+Template.taggedCourses.rendered = function () {
+  // Get reference to template instance
+  var instance = this;
+
+  // Set the page site title for SEO
+  Meta.setTitle('Courses tagged "' + instance.tag + '"');
+};
+
 Template.taggedCourses.helpers({
     'courses': function () {
-        return Courses.find().fetch();
+      // Get reference to template instance
+      var instance = Template.instance();
+
+      // Get tag from template instance
+      var tag = instance.tag;
+
+      // Fetch courses matching current tag
+      var taggedCourses = Courses.find({"keywords": tag}).fetch();
```

Comment:
Thanks for your Work, @brylie! We should add another issue for fixing the
subscriptions on a global scale. I am currently not at the computer but
will check later

Am Mittwoch, 27. Januar 2016 schrieb Brylie Christopher Oxley :

> In client/templates/courses/tagged-courses.js
> <https://github.com/Crowducate/crowducate-platform/pull/144#discussion_r50961646>
> :
>
> >  Template.taggedCourses.helpers({
> >      'courses': function () {
> > -        return Courses.find().fetch();
> > +      // Get reference to template instance
> > +      var instance = Template.instance();
> > +
> > +      // Get tag from template instance
> > +      var tag = instance.tag;
> > +
> > +      // Fetch courses matching current tag
> > +      var taggedCourses = Courses.find({"keywords": tag}).fetch();
>
> I added a "keywords" filter to the query, so that it will only return
> courses where keywords contains tags. This ensures that the page will show
> only tagged courses, despite other active subscriptions.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Crowducate/crowducate-platform/pull/144/files#r50961646>
> .
>